modules/thermalmanager.c

```diff
@@ -52,545 +156,995 @@
-              (new_config.state[i].min >= new_config.state[i].max) ||
-              ((i > THERMAL_STATUS_LOW) && (new_config.state[i].min <= new_config.state[i-1].min)) ||
-              ((i > THERMAL_STATUS_LOW) && (new_config.state[i].max <= new_config.state[i-1].max)) ||
-              ((i > THERMAL_STATUS_LOW) && (new_config.state[i].min > new_config.state[i-1].max)) ||
-              (new_config.state[i].maxtime < 10) ||
-              (new_config.state[i].maxtime > 1000)) {
-              success = false;
-          }
-      }
-      if (success) {
-          /* Note, it is important to give big enough window min..max
-           * then IPHB can freely choose best wake-up time
-           */
-          new_config.state[i].mintime = new_config.state[i].maxtime/2;
-      } else {
-          dsme_log(LOG_ERR, "thermalmanager: syntax error in thermal tuning on line %d", i+1);
-          break;
-      }
-  }
-  if (success) {
-      *config = new_config;
-  }
-
-  return success;
+    thermal_manager_handle_temperature_query(req, "battery", rsp);
 }
 
-static void thermal_object_try_to_read_config(thermal_object_t* thermal_object)
+/* Handle com.nokia.thermalmanager.sensor_temperature D-Bus method call
+ *
+ * Provides backwards compatibility with legacy D-Bus method
+ * that was originally implemented in Harmattan specific
+ * sensor backend plugin.
+ *
+ * @param req   D-Bus method call message
+ * @param rsp   Where to store D-Bus method return message
+ */
+static void
+thermal_manager_get_sensor_temperature_cb(const DsmeDbusMessage *req,
+                                          DsmeDbusMessage **rsp)
 {
-  FILE* f;
-
-  if ((f = thermal_tuning_file(thermal_object->conf->name))) {
-
-      if (thermal_object_config_read(thermal_object->conf, f)) {
-          dsme_log(LOG_NOTICE,
-                   "thermalmanager: Read thermal tuning file for %s",
-                   thermal_object->conf->name);
-      } else {
-          dsme_log(LOG_NOTICE,
-                   "thermalmanager: Thermal tuning file for %s discarded. Using default values",
-                   thermal_object->conf->name);
-      }
-
-      fclose(f);
-  } else {
-      dsme_log(LOG_NOTICE,
-               "thermalmanager: No thermal tuning file for %s. Using default values",
-               thermal_object->conf->name);
-  }
+    const char *sensor = dsme_dbus_message_get_string(req);
+    thermal_manager_handle_temperature_query(req, sensor, rsp);
 }
 
-static thermal_object_t* thermal_object_copy(
-  const thermal_object_t* thermal_object)
+/** Array of D-Bus method calls supported by this plugin */
+static const dsme_dbus_binding_t dbus_methods_lut[] =
 {
-  thermal_object_t*               copy_object;
-  thermal_object_configuration_t* copy_config;
+    { thermal_manager_get_thermal_state_cb,       thermalmanager_get_thermal_state },
+    { thermal_manager_get_surface_temperature_cb, "estimate_surface_temperature" },
```

Comment:
Do we have named constants for `"estimate_surface_temperature"` and the others? Maybe in public headers even?